clang/tools/clang-linker-wrapper/OffloadWrapper.h

```diff
@@ -25,4 +30,49 @@ llvm::Error wrapCudaBinary(llvm::Module &M, llvm::ArrayRef<char> Images);
 /// registers the images with the HIP runtime.
 llvm::Error wrapHIPBinary(llvm::Module &M, llvm::ArrayRef<char> Images);
 
+// SYCL binary image formats supported.
+enum class SYCLBinaryImageFormat {
+  BIF_None,   // Undetermined Image kind
+  BIF_Native, // Native Image kind
+  BIF_SPIRV,  // SPIR-V
+  BIF_LLVMBC  // LLVM bitcode
+};
+
+SYCLBinaryImageFormat getBinaryImageFormatFromString(llvm::StringRef S);
+
+// TODO: add constructors/utils for creating these images in
+// clang-linker-wrapper
+struct SYCLImage {
+  std::string Image;
+  std::optional<llvm::util::PropertySetRegistry> PropertyRegistry;
+
+  // Note: Must be null-terminated.
+  std::optional<llvm::StringRef> Entries;
+
+  // Offload target triple. TODO: check is not performed yet.
+  std::string Target;
+  // format of the image data - SPIRV, LLVMIR bitcode, etc
+  SYCLBinaryImageFormat Format =
+      SYCLBinaryImageFormat::BIF_None; // TODO: is not used in Clang Driver.
+                                       // legacy
+  // target/compiler specific options what are suggested to use to "compile"
+  // program at runtime.
+  //  Allowed to be empty.
+  std::string CompileOptions;
+  // Target/Compiler specific options that are suggested to use to "link"
+  // program at runtime. Allowed to be empty.
+  std::string LinkOptions;
+};
+
+SYCLImage getSYCLImage();
+
+struct SYCLWrappingOptions {
+  bool EmitRegistrationFunctions = true;
+};
+
+// TODO: check whether we need to support -fiopenmp Mode.
```

Comment:
Adding @RaviNarayanaswamy for comment here. Thanks

Reply:
We need to support openmp if that is your question.  If you question is between -fopenmp and -fiopenmp then there should be no difference at this stage.

Reply:
OpenMP support in clang-linker-wrapper and clang-offload-wrapper diverges significantly. Just want to make sure that we do not use community approach for OpenMP and introduce one additional for our use case.
@RaviNarayanaswamy Can you confirm my conclusion that we need to stick to our implementation of offloading and wrapping?

Reply:
I checked implementations again and it looks like our wrapping functionality of OpenMP is exactly the same as community one. It seems that we can use community version for OpenMP.

Reply:
> Just want to make sure that we do not use community approach for OpenMP and introduce one additional for our use case.

Unless there is a strong case for it, we should *not* diverge from the community approach, I would say. It only complicates maintenance

Reply:
Our internal customization for OpenMP diverges significantly.